src/notebookrenderer/index.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  createRendermimePlugin,
+  JupyterFrontEnd,
+  JupyterFrontEndPlugin
+} from '@jupyterlab/application';
+import { ISessionContext, WidgetTracker } from '@jupyterlab/apputils';
+import { IChangedArgs } from '@jupyterlab/coreutils';
+import { MimeDocument } from '@jupyterlab/docregistry';
+import { INotebookTracker } from '@jupyterlab/notebook';
+import { IRenderMimeRegistry } from '@jupyterlab/rendermime';
+import { IRenderMime } from '@jupyterlab/rendermime-interfaces';
+import { IKernelConnection } from '@jupyterlab/services/lib/kernel/kernel';
```

Comment:
```suggestion
import { IKernelConnection } from '@jupyterlab/services';
```

I recall having dedupe issues when doing this kind of nested imports some time ago. Is `IKernelConnection` not exported by `@jupyterlab/services`?

Reply:
Indeed, this comes from the auto-import feature of VS code, I should have used `Kernel.IKernelConnection` instead.